internal/pkg/tfconf/module.go

```diff
@@ -122,20 +126,57 @@ func CreateModule(options *Options) (*Module, error) {
 		}
 	}
 
+	options.OutputValuesPath = os.Getenv("FOO")
+	// if OutputValuesPath is empty, set the default and generate the file
+	if options.OutputValues && options.OutputValuesPath == "" {
+		err := os.Setenv("FOO", "terraform-outputs.json")
+		if err != nil {
+			log.Fatal(err)
+		}
+		_, err = exec.Command("cd", options.Path, "&&", "terraform", "output", "--json", ">>", options.OutputValuesPath).Output()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	var terraformOutputs map[string]*TerraformOutput
+	if options.OutputValues {
+		byteValue, err := ioutil.ReadFile(options.OutputValuesPath)
+		if err != nil {
+			return nil, fmt.Errorf("caught error while reading the terraform outputs file at %s: %v", options.OutputValuesPath, err)
+		}
+		if err := json.Unmarshal(byteValue, &terraformOutputs); err != nil {
+			return nil, err
+		}
+	}
+
+	// output module
 	var outputs = make([]*Output, 0, len(mod.Outputs))
 	for _, output := range mod.Outputs {
 		outputDescription := output.Description
 		if outputDescription == "" {
 			outputDescription = readComment(output.Pos.Filename, output.Pos.Line-1)
 		}
-		outputs = append(outputs, &Output{
-			Name:        output.Name,
-			Description: String(outputDescription),
-			Position: Position{
-				Filename: output.Pos.Filename,
-				Line:     output.Pos.Line,
-			},
-		})
+		if options.OutputValues {
+			outputs = append(outputs, &Output{
+				Name:        output.Name,
+				Description: String(outputDescription),
+				Value:       terraformOutputs[output.Name].Value,
+				Position: Position{
+					Filename: output.Pos.Filename,
+					Line:     output.Pos.Line,
+				},
+			})
+		} else {
+			outputs = append(outputs, &Output{
+				Name:        output.Name,
+				Description: String(outputDescription),
+				Position: Position{
+					Filename: output.Pos.Filename,
+					Line:     output.Pos.Line,
+				},
+			})
+		}
 	}
```

Comment:
This looks cleaner and more readable to me:

```
	for _, o := range mod.Outputs {
		outputDescription := o.Description
		if outputDescription == "" {
			outputDescription = readComment(o.Pos.Filename, o.Pos.Line-1)
		}
		output := &Output{
			Name:        o.Name,
			Description: String(outputDescription),
			Position: Position{
				Filename: o.Pos.Filename,
				Line:     o.Pos.Line,
			},
		}
		if options.OutputValues {
			output.Value = terraformOutputs[output.Name].Value
		}
		outputs = append(outputs, output)
	}
```

Reply:
Thanks!